opal/corelib/constants.rb

```diff
@@ -1,4 +1,5 @@
-::RUBY_PLATFORM       = 'opal'
+# backtick_javascript: true
+::RUBY_PLATFORM       = `Opal.platform.ruby_platform`
```

Comment:
From the gist of it, it looks like a breaking change?

Reply:
This is required so that ruby specs correctly can identify if they are running on Windows or Linux and select platform specific specs. So in case of the node driver, this returns 'opal win' on Windows or 'opal linux' on Linux or 'opal' on others. Other drivers may return 'opal macos' or something.

Reply:
Platform OS identification usually happens by regexp matching RUBY_PLATFORM against /win/, /linux/ etc.

Reply:
What I mean is opal_platform_check rewriter which strips the code based on where it's used. And currently Opal code may use `==` to check if something is running on Opal, along with RUBY_ENGINE. This is what we will need to investigate more thoroughly.

Reply:
I add it to TODO